### PR TITLE
Allow player authentication in leaderboard

### PR DESCRIPTION
**Pull request by 4ian**
Games (leaderboards):
- Allow passing the connected player to the leaderboard iframe
- Allow communication with the leaderboard iframe to:
  - Get a request to authenticate the player
  - Communicate the result of it
- Fix display of leaderboard iframes in development on iOS
- Add a setting (that can be disabled with an action) to prefer to send a score for the connected player (even if the action to send as an anonymous player is used).

Games (authentication):
- Fix iOS app authentication (postMessage does not work in a Cordova In App Browser, use websocket instead).
- Rework authentication so that the caller code can know when it's done.
- Allow the action that opens the authentication window to wait for the authentication to be done (or closed)

Editor:
- Add option to avoid displaying login buttons in the leaderboards

### CODE REVIEW

File: GDJS/Runtime/Cordova/www/index.html [hunk 1]
```diff
@@ -8,7 +8,7 @@
 		/* Prevent copy paste for all elements except text fields */
 		*  { -webkit-user-select:none; -webkit-tap-highlight-color:rgba(255, 255, 255, 0); }
 		input, textarea  { -webkit-user-select:text; }
-		body { background-color:black; color:white }
```
Comment (4ian) [inline GDJS/Runtime/Cordova/www/index.html:11]: This was useless and cascading inside the authentication window messages.

File: Extensions/Leaderboards/JsExtension.js [hunk 2]
```diff
@@ -92,6 +94,28 @@ module.exports = {
...
+      )
+      .addCodeOnlyParameter('currentScene', '')
+      .addParameter('yesorno', _('Enable?'), '', false)
+      .setHelpPath('/all-features/leaderboards')
+      .getCodeExtraInformation()
+      .setIncludeFile('Extensions/Leaderboards/leaderboardstools.js')
+      .setFunctionName(
+        'gdjs.evtTools.leaderboards.setPreferSendConnectedPlayerScore'
+      );
+    // TODO: add player authentication files.
```
Comment (ClementPasteau) [inline Extensions/Leaderboards/JsExtension.js:117]: You should!

File: Extensions/Leaderboards/JsExtension.js [hunk 3]
```diff
@@ -92,6 +94,28 @@ module.exports = {
         'gdjs.evtTools.leaderboards.saveConnectedPlayerScore'
       );
 
+    extension
+      .addAction(
+        'SetPreferSendConnectedPlayerScore',
+        _('Enable the score to the connected player'),
```
Comment (ClementPasteau) [inline Extensions/Leaderboards/JsExtension.js:100]: this doesn't make sense. `Enable attaching the score to the connected player`?

File: Extensions/Leaderboards/JsExtension.js (hunk 2 above)
Comment (4ian) [reply Extensions/Leaderboards/JsExtension.js]: Fixed

File: Extensions/Leaderboards/JsExtension.js (hunk 3 above)
Comment (4ian) [reply Extensions/Leaderboards/JsExtension.js]: Fixed

**Pull request merged**